dotcom-rendering/src/components/ReaderRevenueLinks.importable.tsx

```diff
@@ -233,7 +233,7 @@ const ReaderRevenueLinksRemote = ({
 					'rr-header-links',
 				);
 			});
-	}, [countryCode]);
+	}, [countryCode, isSignedIn]);
```

Comment:
I think there may be a subtle difference between this case and the case in `SupportTheG`?


In this case `isSignedIn` is `boolean` rather than `boolean | undefined`, and will be `false` when the signed in status is "pending". Then, once we have confirmed the user has signed in, it will switch to `true` (it will remain `false` if they're signed out).


`useOnce` waits if any dependency is `undefined`, and then runs once, and only once, when it confirms that none of them are `undefined`. But this dependency will _never_ be `undefined`, because it's just `boolean`. So in the "pending" case `useOnce` will still run with `isSignedIn` set to `false`. Then, when we confirm the user has signed in, it will switch to `true`. But `useOnce` won't run again because it only runs once.

I think if we want to use `isSignedIn` in the `useOnce` dependency array we have to do something similar to the `SupportTheG` behaviour, i.e. use something like the `getIsSignedIn` function and allow `isSignedIn` to be `undefined` in the "pending" case? That way `useOnce` will wait to run until we know for sure the user is either signed in or signed out.

FYI @Georges-GNM

Reply:
Thanks - I will look into this.

It looked to me like the Header from SDC is only being generated by the using the SupportTheG file now, so maybe the purchasing info targeting doesn't even need to be sent by the Reader Revenue links anymore.

Reply:
Yes I agree - the remote logic can be removed from `ReaderRevenueLinks.importable.tsx` because it has a check for `inHeader`, and I can see it's only being imported by the footer now!
But let's not let that slow down getting this fix out